File: integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/model/CarVEntity.java
package io.quarkus.it.mongodb.panache.model;

import java.util.Objects;

import io.quarkus.mongodb.panache.common.MongoEntity;
import io.quarkus.mongodb.panache.common.Version;

@MongoEntity
public class CarVEntity extends VehicleEntity {

    @Version
    public Long version;

    public CarVEntity() {
    }

    public CarVEntity(String id, Integer modelYear, Long version) {
        super(id, modelYear);
        this.version = version;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (!(o instanceof CarVEntity))
            return false;
        if (!super.equals(o))
            return false;
        CarVEntity carV = (CarVEntity) o;
        return Objects.equals(version, carV.version);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), version);
    }
}

File: integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/model/CarVEntity.kt
package io.quarkus.it.mongodb.panache.model

import io.quarkus.mongodb.panache.common.MongoEntity
import io.quarkus.mongodb.panache.common.Version
import org.bson.codecs.pojo.annotations.BsonCreator
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.codecs.pojo.annotations.BsonProperty

@MongoEntity
class CarVEntity @BsonCreator constructor(
    @BsonId override var id: String,
    @BsonProperty("modelYear") override var modelYear: Int?,
    @BsonProperty("version") @Version
    var version: Long?
) : VehicleEntity(id, modelYear)
